src/lib/TextArea/TextArea.svelte

```
@@ -0,0 +1,223 @@
+    const next = Math.min(Math.max(el.scrollHeight, minHeight), maxHeight);
+    el.style.height = `${next}px`;
+    el.style.overflowY = el.scrollHeight > maxHeight ? 'auto' : 'hidden';
+  }
+
+  $effect(() => {
+    // Re-run whenever the value changes (and on mount) while autoResize is enabled.
+    void value;
+    if (autoResize) {
+      adjustHeight();
+    }
+  });
```

ESLint

[error] $effect is not allowed. Use a different reactive pattern instead.

(no-restricted-syntax)